File: picpay/app/src/main/java/br/com/picpay/picpay/model/Transaction.java
package br.com.picpay.picpay.model;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import br.com.picpay.core.util.StringUtil;

public class Transaction {

    private String id;
    private int timestamp;
    private float value;
    private User destinationUser;
    private boolean success;
    private String status;

    @NonNull
    public String getId() {
        return StringUtil.fixString(id);
    }

    public int getTimestamp() {
        return timestamp;
    }

    public float getValue() {
        return value;
    }

    @Nullable
    public User getDestinationUser() {
        return destinationUser;
    }

    public boolean isSuccess() {
        return success;
    }

    @NonNull
    public String getStatus() {
        return StringUtil.fixString(status);
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setTimestamp(int timestamp) {
        this.timestamp = timestamp;
    }

    public void setValue(float value) {
        this.value = value;
    }

    public void setDestinationUser(User destinationUser) {
        this.destinationUser = destinationUser;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}

File: picpay/app/src/main/java/br/com/picpay/picpay/model/Transaction.kt
package br.com.picpay.picpay.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

class Transaction(var id: Int = 0,
                  var timestamp: Long = 0,
                  var value: Float = 0F,
                  @SerializedName("destination_user")
                  var destinationUser: User? = null,
                  var success: Boolean = false,
                  var status: String? = "",
                  var cardCompany: String? = "",
                  var cardLastNumbers: String? = ""): Parcelable {

    constructor(parcel: Parcel) : this(
        parcel.readInt(),
        parcel.readLong(),
        parcel.readFloat(),
        parcel.readParcelable(User::class.java.classLoader),
        parcel.readByte() != 0.toByte(),
        parcel.readString(),
        parcel.readString(),
        parcel.readString()
    ) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(id)
        parcel.writeLong(timestamp)
        parcel.writeFloat(value)
        parcel.writeParcelable(destinationUser, flags)
        parcel.writeByte(if (success) 1 else 0)
        parcel.writeString(status)
        parcel.writeString(cardCompany)
        parcel.writeString(cardLastNumbers)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Transaction> {
        override fun createFromParcel(parcel: Parcel): Transaction {
            return Transaction(parcel)
        }

        override fun newArray(size: Int): Array<Transaction?> {
            return arrayOfNulls(size)
        }
    }
}